p2p/peer.py

```diff
@@ -693,6 +696,16 @@ def subscribe(self, subscriber: PeerPoolSubscriber) -> None:
             raise Exception(
                 "We cannot handle multiple subscribers yet; "
                 "see https://github.com/ethereum/py-evm/issues/768 for details")
+
+        """
+        There is max_peers defined from CLI.
+        We drop the subscriber if it's already maxed out.
+        """
+        if len(self.connected_nodes) >= self.max_peers:
+            raise HandshakeFailure(
+                "Not processing the handshake because too many peers. "
+                "Max peers %d", self.max_peers)
```

Comment:
This is not the correct place to raise `HandshakeFailure` as there's no handshake going on here; this is the `PeerPool.subscribe()` method. Or am I missing something?